tests/integration/models/application.spec.coffee

```diff
@@ -35,43 +35,87 @@ describe 'Application Model', ->
 
 		describe 'resin.models.application.create()', ->
 
-			it 'should be able to create an application', ->
-				resin.models.application.create('FooBar', 'raspberry-pi').then ->
+
+			it 'should be able to create an application w/o providing an application type', ->
+				resin.models.application.create
+					name: 'FooBar'
+					deviceType: 'raspberry-pi'
+				.then ->
 					promise = resin.models.application.getAll()
 					m.chai.expect(promise).to.eventually.have.length(1)
 
 			it 'should be able to create a child application', ->
-				resin.models.application.create('FooBar', 'raspberry-pi').then (parentApplication) ->
-					resin.models.application.create('FooBarChild', 'generic', parentApplication.id)
+				resin.models.application.create
+					name: 'FooBar'
+					applicationType: 'microservice-starter'
+					deviceType: 'raspberry-pi'
+				.then (parentApplication) ->
+					resin.models.application.create
+						name: 'FooBar'
+						deviceType: 'generic'
+						parent: parentApplication.id
 				.then ->
 					resin.models.application.getAll()
 				.then ([ parentApplication, childApplication ]) ->
 					m.chai.expect(childApplication.depends_on__application.__id).to.equal(parentApplication.id)
 
+			it 'should be able to create an application with a specific application type', ->
+				resin.models.application.create
+					name: 'FooBar'
+					applicationType: 'microservice-starter'
+					deviceType: 'raspberry-pi'
+				.then ->
+					promise = resin.models.application.getAll()
+					m.chai.expect(promise).to.eventually.have.length(1)
+
+			it 'should be rejected if the application type is invalid', ->
+				promise = resin.models.application.create
+					name: 'FooBar'
+					applicationType: 'non-existing'
+					deviceType: 'raspberry-pi'
+				m.chai.expect(promise).to.be.rejectedWith('Invalid application type: foobarbaz')
```

Comment:
Surely this test fails? That's not the right app type in the error here.

Reply:
You caught me...
I guess we will have to wait for the respective API branch to be deployed on staging in order to verify that all tests indeed pass.

Reply:
You should be able to run the API branch locally yourself in the devenv and test against that pretty easily. Ping me in FD if you want a hand.